source/Src/Unity-CoreClr/UnityContainer.cs

```diff
@@ -478,8 +480,12 @@ public void Dispose()
         /// method, false if being called from a finalizer.</param>
         protected virtual void Dispose(bool disposing)
         {
-            if (disposing)
+            if (disposing && !inDispose)
             {
+                // Avoid infinite loop when someone
+                //  registers something which would end up 
+                //  disposing this container (e.g. container.RegisterInsance(container))
+                inDispose = true;
                 if (lifetimeContainer != null)
                 {
                     lifetimeContainer.Dispose();
```

Comment:
is lifetimecontainer.dispose() the source of the stack overflow?  Maybe nulling it out first and then calling dispose could produce the same results without the need of having an extra "indispose" Boolean private field.  (Obviously you'd need a local var to hold lifetimecontainer)